clients/fides-js/src/lib/providers/event-context.tsx

```diff
@@ -0,0 +1,94 @@
+import { createContext, h } from "preact";
+import { MutableRefObject, ReactNode } from "preact/compat";
+import { useCallback, useContext, useMemo, useRef } from "preact/hooks";
+
+import { FidesCookie, ServingComponent } from "../consent-types";
+import {
+  dispatchFidesEvent,
+  FidesEventDetailsTrigger,
+  FidesEventExtraDetails,
+  FidesEventType,
+} from "../events";
+
+interface UseEventProps {
+  triggerRef: MutableRefObject<FidesEventDetailsTrigger>;
+  setTrigger: (newValue: FidesEventDetailsTrigger) => void;
+  servingComponentRef: MutableRefObject<ServingComponent | undefined>;
+  setServingComponent: (newValue: ServingComponent | undefined) => void;
+  dispatchFidesEventAndClearTrigger: (
+    type: FidesEventType,
+    fidesCookie: FidesCookie | undefined,
+    extraDetails?: FidesEventExtraDetails,
+  ) => void;
+}
+
+const EventContext = createContext<UseEventProps | Record<any, never>>({});
+
+export const EventProvider = ({ children }: { children: ReactNode }) => {
+  const triggerRef = useRef<FidesEventDetailsTrigger>();
+  const servingComponentRef = useRef<ServingComponent | undefined>();
```

Comment:
We really only need to set this when the banner and modal are shown, and since it's stored in the context, we don't need to keep setting it for each event. We do clear this when the overlays are closed to keep it honest.